### DISCUSSION TITLE
How to stop a process when a job is discarded?

### DISCUSSION THREAD

**Discussion by BrunoQuaresma** [OptimalBits/bull#2279] [Q&A]
Hey folks, I have an option where the user can cancel a job and what I'm doing right now is to use `job.discard()`. I don't know if this is the best method, I also see there is a `remove()`. But right after discarding or removing a job, I would like to stop the process that was initialized which in my case is an FFmpeg transcoding which is pretty heavy. Thanks!

**Comment by BrainBust** [comment]
Same situation: neither `discard()`, `remove()` or `moveToCompleted()` seem to have any effect.

What's the counter part of `add()`? 
I can start a job but i am unable to stop a running one, which should be a very basic method in any queue library.

If you try and stop multiple jobs by using job status on an interval in quick succession , some do stop after returning `resolve()`, `reject()` or `cb()`, but at random certain jobs simply linger and turn into a zombie state where they never stop after `cb()` was returned.

for instance:

In a controller

```
const job = await this.queue.getJob(id);
if (job) {
  await job.update({ ...job.data, stop: true });
}
```

In a processor
```
this.interval = setInterval(async () => {
  const jobCheck = await this.queue.getJob(job.id);
  if (jobCheck.data.stop) {
    clearInterval(this.interval);
    return Promise.resolve(true)
  }
}, 3000)
```

The above code has no guarantee the job will terminate, certainly not if multiple jobs need to stop concurrently, some just turn into zombie processes.
